iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -862,14 +862,14 @@ public void merge(Network other) {
         }
 
         // try to find dangling lines couples
-        Map<String, DanglingLine> dl1byXnodeCode = new HashMap<>();
-        for (DanglingLine dl1 : getDanglingLines()) {
-            if (dl1.getUcteXnodeCode() != null) {
-                dl1byXnodeCode.put(dl1.getUcteXnodeCode(), dl1);
-            }
-        }
         List<MergedLine> lines = new ArrayList<>();
         for (DanglingLine dl2 : Lists.newArrayList(other.getDanglingLines())) {
+            Map<String, DanglingLine> dl1byXnodeCode = new HashMap<>();
```

Comment:
Is it normal the map 'dl1byXnodeCode' would be re-generated every time for dl2?

Reply:
It was the bug. As dangling line might have been removed at each iteration map need to be regenerated otherwise the map could contain a removed dangling line.